src/web/components/SubNav/SubNav.tsx

```diff
@@ -1,87 +1,207 @@
+`;
+
+const subnavCollapsed = css`
+    ${subnav};
+    max-width: calc(100% - 60px);
 
-        const lis = ul.querySelectorAll('li');
-        const lastLi = lis[lis.length - 1];
+    ${from.mobileLandscape} {
+        max-width: calc(100% - 70px);
+    }
+`;
 
-        const ulTop = ul.getBoundingClientRect().top;
-        const liTop = lastLi.getBoundingClientRect().top;
+const fontStyle = css`
+    ${textSans.medium()};
+    font-weight: 500;
+    color: ${palette.neutral[7]};
+    padding: 0 5px;
+    height: 36px;
+    /* Design System: Line height is being used here for centering layout, we need the primitives */
+    line-height: 36px;
 
-        return ulTop !== liTop;
+    ${from.tablet} {
+        height: 42px;
+        /* Design System: Line height is being used here for centering layout, we need the primitives */
+        /* stylelint-disable-next-line property-blacklist */
+        line-height: 42px;
     }
+`;
 
-    public render() {
-        const { subnav, pillar, currentNavLink } = this.props;
+const linkStyle = css`
+    ${fontStyle};
+    float: left;
+    text-decoration: none;
 
-        if (!subnav) {
-            return null;
+    :hover {
+        text-decoration: underline;
+    }
+`;
+const selected = css`
+    font-weight: 700;
+`;
+const moreStyle = css`
+    ${fontStyle};
+
+    cursor: pointer;
+    border: none;
+    background-color: transparent;
+    color: ${text.supporting};
+
+    :hover {
+        color: ${palette.news.main};
+    }
+
+    ${from.desktop} {
+        display: none;
+    }
+`;
+
+const parentLinkStyle = css`
+    ${linkStyle};
+    font-weight: 700;
+`;
+const ps1 = css`
+    :after {
+        content: '';
+        display: inline-block;
+        width: 0;
+        height: 0;
+        border-top: 6px solid transparent;
+        border-bottom: 6px solid transparent;
+        border-left: 10px solid ${palette.neutral[7]};
+        margin-top: 12px;
+        margin-left: 2px;
+
+        ${from.tablet} {
+            margin-top: 16px;
+        }
+    }
+`; // I'm not sure what the palette.neutral is for this should always receive a pillar by types.
+const psp = pillarMap(
+    pillar => css`
+        :after {
+            border-left-color: ${pillarPalette[pillar].main};
+        }
+    `,
+);
+
+const trimLeadingSlash = (url: string): string =>
+    url.substr(0, 1) === '/' ? url.slice(1) : url;
+
+export const SubNav = ({ subNavSections, pillar, currentNavLink }: Props) => {
+    const [showMore, setShowMore] = useState(false);
+    const [isExpanded, setIsExpanded] = useState(false);
+    const ulRef = useRef<HTMLUListElement>(null);
+
+    useEffect(() => {
+        const ulEl = ulRef.current;
+        if (ulEl) {
+            const lis = ulEl.querySelectorAll('li');
+            const lastLi = lis[lis.length - 1];
+
+            const ulTop = ulEl.getBoundingClientRect().top;
+            const liTop = lastLi.getBoundingClientRect().top;
+
+            setShowMore(ulTop !== liTop);
```

Comment:
I was initially confused what this was doing but read through it and it's actually quite a neat solution. It's a shame to have 2 `getBoundingClientRect` calls though, they're pretty expensive. I honestly can't think of a way off the top of my head but if while you're in the code you can see a way to refactor this to a more optimal method... Maybe `offsetTop`?

Reply:
thanks for pointing it out, I will try and find a cleaner solution

Reply:
after looking at it I cant think of a better solution so I am keeping it as is